src/server/player_sao.cpp

```diff
@@ -485,6 +485,13 @@ u16 PlayerSAO::punch(v3f dir,
 	return hitparams.wear;
 }
 
+void PlayerSAO::rightClick(ServerActiveObject *clicker)
+{
+	if (isAttached())
```

Comment:
Why does this depend on the attached state?
Players can always be right-clicked, thus it might be better to allow the callback to be run in every case.

Reply:
> Why does this depend on the attached state?
> Players can always be right-clicked, thus it might be better to allow the callback to be run in every case.

Only if "pointable = true" <-- object property

Reply:
There was actually a good reason for this, but I can't recall. I remember when I didn't check the attached state, things went awry which might explain why the right-click handler for LuaEntitySAO's includes the same check. I'll have to investigate.

Reply:
Actually I stand corrected, it was the on deactivate handler. So I should be able to remove the conditional.

Reply:
No, I loved the on_rightclick think!

:-(